### PR TITLE
Add iter_mut on arena

### PR DESCRIPTION
**Pull request by seifane**
Added a way to retreive a mutable iterator on nodes.

### CODE REVIEW

File: src/arena.rs [hunk 1]
```diff
@@ -214,6 +214,10 @@ impl<T> Arena<T> {
         self.nodes.iter()
     }
 
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = &mut Node<T>> {
```
Comment (saschagrunert) [inline src/arena.rs:217]: Can you please add a doc comment for this? :)

**Review by saschagrunert — COMMENTED** [review]
Thanks! I just have a nit on the docs.

File: src/arena.rs (hunk 1 above)
Comment (seifane) [reply src/arena.rs:217]: Do you want the doc with example ? :]

Comment (saschagrunert) [reply src/arena.rs:217]: That would be wonderful!

**Review by saschagrunert — APPROVED** [review]
Thanks!

**Comment by saschagrunert** [comment]
Let's fix the CI:

```
Checking indextree v4.3.1 (/home/runner/work/indextree/indextree)
error: this expression borrows a reference (`&arena::Arena<T>`) that is immediately dereferenced by the compiler
    --> src/id.rs:[10](https://github.com/saschagrunert/indextree/runs/5274470991?check_suite_focus=true#step:6:10)23:30
     |
1023 |                 id.ancestors(&arena) // traverse ancestors upwards
     |                              ^^^^^^ help: change this to: `arena`
     |
     = note: `-D clippy::needless-borrow` implied by `-D warnings`
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
```

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**